crates/chain/src/tx_graph.rs

```diff
@@ -761,6 +749,42 @@ impl<A: Clone + Ord> TxGraph<A> {
     }
 }
 
+/// An update to [`TxGraph`].
+pub struct Update<A> {
+    whole_txs: HashMap<Txid, Arc<Transaction>>,
+    partial_txs: HashMap<OutPoint, TxOut>,
+    last_seen: HashMap<Txid, u64>,
+    anchors: BTreeSet<(A, Txid)>,
+}
```

Comment:
Remember that we will be moving `Update` into a separate crate `bdk_core` so we won't have access to private members.

Additionally, we don't want to provide floating outputs to our update logic when we already have the full tx.

**With these two requirements in mind, let's see various solutions.**

* Publicize all members. This is a bad idea, since we can't enforce our invariances.
* Enforce invariance when caller inserts data. When inserting full tx, rm all associated floating txouts of the tx's txid. Don't insert floating txout when there is a full tx of same txid.
* Enforce invariance when extracting data. When extracting full txs, also rm floating txouts with the same txid. When extracting floating txouts, ignore elements which have an associated full tx.

I think enforcing invariance when caller inserts data is easier to reason with.